### PR TITLE
feat: adding fairea tool

### PR DESCRIPTION
**Pull request by fracarfer5**
adding fairness accuracy trade-off tool, implemented main class, tutorial and test units

### CODE REVIEW

File: holisticai/utils/trade_off_analysers/fairea.py [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+from .utils_fairea import (
+    do_line_segments_intersect,
+    get_area,
+    get_baseline_bounds,
+    line,
+)
+
+
+class Fairea:
+    def __init__(
```
Comment (crismunoz) [inline holisticai/utils/trade_off_analysers/fairea.py:49]: - Follow Bias Mitigator format.
- Reference (URL)
- explain acc_metric (used for what?)

File: holisticai/utils/trade_off_analysers/fairea.py [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+            self.fair_fn = statistical_parity
+        elif fair_metric == "aod":
+            self.fair_fn = average_odds_diff
+        self.fair_metric = fair_metric
+        self.methods = dict()
+        self.normalized_methods = dict()
+        self.best = None
+        self.mitigation_regions = dict()
+
+    def create_baseline(self, x, y, group_a, group_b):
```
Comment (crismunoz) [inline holisticai/utils/trade_off_analysers/fairea.py]: Use this format:
fairea.create_baseline(X_train, y_train, group_a_tr, group_b_tr, num_splits=10, test_size=0.2)

File: tutorials/trade_off/fairea_tutorial.ipynb
Comment (crismunoz) [inline tutorials/trade_off/fairea_tutorial.ipynb:1]: explain : plot_baseline
use figsize instead of size
# Fill area good-trade of and bad trade-off
fairea.region_classification() # trade_off_region_classification
fairea.determine_area() # trade_off_region_area
fairea.get_best() # get_best_model

File: holisticai/utils/trade_off_analysers/fairea.py [hunk 3]
```diff
@@ -0,0 +1,538 @@
...
+    do_line_segments_intersect,
+    get_area,
+    get_baseline_bounds,
+    line,
+)
+
+
+class Fairea:
+    """
+    Fairea class for calculating the trade-off between accuracy and fairness of a classification model.
```
Comment (crismunoz) [inline holisticai/utils/trade_off_analysers/fairea.py]: only binary?

**Review by crismunoz — CHANGES_REQUESTED** [review]

File: holisticai/utils/trade_off_analysers/fairea.py [hunk 4]
```diff
@@ -0,0 +1,538 @@
...
+                    average odds difference (aod)
+        acc_metric: string
+                    accuracy metric to be used in the baseline creation and trade-off comparison: accuracy (acc) or roc-auc (auc)
+
+        verbose: boolean
+                whether to print the progress or not
+
+        """
+        self.verbose = verbose
+        if acc_metric == "acc":
```
Comment (crismunoz) [inline holisticai/utils/trade_off_analysers/fairea.py]:
```python
metrics = { 
  "acc": accuracy_score,
  "auc":  roc_auc_score,
  "sp": statistical_parity,
  "aod": average_odds_diff
}

self.acc_fn = metrics [acc_metric]
```

File: holisticai/utils/trade_off_analysers/fairea.py [hunk 5]
```diff
@@ -0,0 +1,538 @@
...
+            data_splits=data_splits,
+            repetitions=repetitions,
+            odds=odds,
+            options=options,
+            degrees=degrees,
+        )
+        self.acc_scaler = MinMaxScaler()
+        self.fair_scaler = MinMaxScaler()
+        self.acc_norm = self.acc_scaler.fit_transform(
+            self.baseline_acc.reshape(-1, 1)
```
Comment (crismunoz) [inline holisticai/utils/trade_off_analysers/fairea.py]: reshape(-1,1) inside __create_baseline

File: holisticai/utils/trade_off_analysers/utils_fairea.py [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+    point : tuple
+        The point to be compared.
+    x_arr : array, shape = (n_samples,)
+        The array of the x axis.
+    y_arr : array, shape = (n_samples,)
+        The array of the y axis.
+    bound : string
+        The bound to be compared if lower or upper.
+
+    """
```
Comment (crismunoz) [inline holisticai/utils/trade_off_analysers/utils_fairea.py:25]: return?

**Review by crismunoz — CHANGES_REQUESTED** [review]
great

File: holisticai/utils/trade_off_analysers/fairea.py (hunk 1 above)
Comment (fracarfer5) [reply holisticai/utils/trade_off_analysers/fairea.py:49]: done

File: holisticai/utils/trade_off_analysers/fairea.py (hunk 4 above)
Comment (fracarfer5) [reply holisticai/utils/trade_off_analysers/fairea.py]: changed

File: holisticai/utils/trade_off_analysers/utils_fairea.py (hunk 6 above)
Comment (fracarfer5) [reply holisticai/utils/trade_off_analysers/utils_fairea.py:25]: done

File: holisticai/utils/trade_off_analysers/fairea.py (hunk 5 above)
Comment (fracarfer5) [reply holisticai/utils/trade_off_analysers/fairea.py]: done

File: holisticai/utils/trade_off_analysers/fairea.py (hunk 2 above)
Comment (fracarfer5) [reply holisticai/utils/trade_off_analysers/fairea.py]: done

**Review by crismunoz — APPROVED** [review]
